### DISCUSSION TITLE
Typescript (Types, Interfaces) not showing up in VSCode

### DISCUSSION THREAD

**Discussion by indicozy** [prisma/prisma#17508] [Q&A] [answered]
Hi everyone,

prisma's interfaces is not showing up in VSCodium IDE. I've got typescript turned on and it's working fine in IDE, but prisma does not show up types.

What am I doing wrong?

Details on the comment below.

![image](https://user-images.githubusercontent.com/51142471/214320016-77b5e44e-3b91-491e-a953-a79e0170e51f.png)

**Comment by indicozy** [comment, marked as answer]
I've found the issue. I'm using a custom prisma module which looks like this:
```ts
import { PrismaClient } from '@prisma/client';

// eslint-disable-next-line import/no-mutable-exports
let prisma;

if (process.env.NODE_ENV === 'production') {
  prisma = new PrismaClient();
} else {
  if (!global.prisma) {
    global.prisma = new PrismaClient();
  }
  prisma = global.prisma;
}
// ... some prisma middlewares
export default prisma;
```

How can I add type safety for custom prisma modules?

Edit: Answer is below, from @nurul3101.

**Reply by nurul3101** [reply]
Can you define the prisma variable like this [example](https://github.com/prisma/prisma-examples/blob/latest/typescript/rest-nextjs-api-routes-auth/lib/prisma.ts)?
It should allow you to have type safety with custom prisma modules.

**Reply by indicozy** [reply]
Yes, thank you!

**Reply by nurul3101** [reply]
No worries! :)

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
